commit b367a1c9d8a0f565effcdb45f86a365be1c8b622
Author: Jan Drzewiecki <jonie@icpnet.pl>
Date:   Fri Aug 13 07:24:29 2010 +0200

    cdrom_id: Drop MEDIA_SESSION_NEXT for DVD-RW-RO
    
    Commit cf2205a fixed the media status for fresh DVD-RW in restricted overwrite
    mode, but missed a detail: We should not report the ID_CDROM_MEDIA_SESSION_NEXT
    property either, since in that mode you can never append tracks/sessions; this
    just works in sequential mode.
    
    Signed-off-by: Martin Pitt <martin.pitt@ubuntu.com>

diff --git a/extras/cdrom_id/cdrom_id.c b/extras/cdrom_id/cdrom_id.c
index d59991c..f9d6e0b 100644
--- a/extras/cdrom_id/cdrom_id.c
+++ b/extras/cdrom_id/cdrom_id.c
@@ -690,8 +690,9 @@ static int cd_media_info(struct udev *udev, int fd)
 	}
 
 determined:
-	/* "other" is e. g. DVD-RAM, can't append sessions there either */
-	if ((header[2] & 3) < 2)
+	/* "other" is e. g. DVD-RAM, can't append sessions there; DVDs in
+	 * restricted overwrite mode can never append, only in sequential mode */
+	if ((header[2] & 3) < 2 && !cd_media_dvd_rw_ro)
 		cd_media_session_next = header[10] << 8 | header[5];
 	cd_media_session_count = header[9] << 8 | header[4];
 	cd_media_track_count = header[11] << 8 | header[6];
